lib/helper/user_comms.rb

```diff
@@ -0,0 +1,25 @@
+module Emerald
+  class UserCommsHelper
+
+    ERROR_INCORRECT_FILE_TYPE = "Oops! Incorrect file type received.\nExpecting a lisp file e.g. example.lisp\n"
+    ERROR_INITIALISE_WITH_STRING_IO = "Error: Initialize with StringIO objects"
+
+    def initialize(stdout)
+      raise ERROR_INITIALISE_WITH_STRING_IO if stdout.nil?
+      @stdout = stdout if stdout.respond_to?(:puts)
```

Comment:
Having argument checks like this isn't very Ruby. Generally you just trust that you're passed an object with the right interface. It's not very safe, but defensive programming is expensive and not great either.

This is def one of the big problems with these dynamic scripting languages.